src/compiler/transformer.ts

```diff
@@ -662,33 +668,51 @@ export function transformNodes<T extends Node>(resolver: EmitResolver | undefine
         }
     }
 }
-
 /** @internal */
-export const nullTransformationContext: TransformationContext = {
-    factory: factory, // eslint-disable-line object-shorthand
-    getCompilerOptions: () => ({}),
-    getEmitResolver: notImplemented,
-    getEmitHost: notImplemented,
-    getEmitHelperFactory: notImplemented,
-    startLexicalEnvironment: noop,
-    resumeLexicalEnvironment: noop,
-    suspendLexicalEnvironment: noop,
-    endLexicalEnvironment: returnUndefined,
-    setLexicalEnvironmentFlags: noop,
-    getLexicalEnvironmentFlags: () => 0,
-    hoistVariableDeclaration: noop,
-    hoistFunctionDeclaration: noop,
-    addInitializationStatement: noop,
-    startBlockScope: noop,
-    endBlockScope: returnUndefined,
-    addBlockScopedVariable: noop,
-    requestEmitHelper: noop,
-    readEmitHelpers: notImplemented,
-    enableSubstitution: noop,
-    enableEmitNotification: noop,
-    isSubstitutionEnabled: notImplemented,
-    isEmitNotificationEnabled: notImplemented,
-    onSubstituteNode: noEmitSubstitution,
-    onEmitNode: noEmitNotification,
-    addDiagnostic: noop,
-};
+export function createTransformationContext(kind: TransformationContextKind.NullContext): NullTransformationContext;

```

Comment:
The overload seems a bit overcomplicated since we'll only ever create one `nullTransformationContext`. You could just leave `nullTransformationContext` as is with the exception of giving it the type `NullTransformationContext` and adding the `kind`, and then just use `return { ...nullTransformationContext, kind: TransformationContext.IsolatedContext, <whatever overrides you want> }`. I realize that's pretty much what you had before, but I think `kind` is actually a bit clearer than the boolean field you were using previously.